File: c_like/premio.cs
using System;

class Program {
    static void Main() {
        // Leitura e conversão das entradas (cada uma em uma linha)
        int p = int.Parse(Console.ReadLine());
        int d = int.Parse(Console.ReadLine());
        int b = int.Parse(Console.ReadLine());

        // Cálculo da pontuação
        int pontuacao = (p * 1) + (d * 2) + (b * 3);

        // Lógica de decisão
        if (pontuacao >= 150) {
            Console.WriteLine("B");
        } else if (pontuacao >= 120) {
            Console.WriteLine("D");
        } else if (pontuacao >= 100) {
            Console.WriteLine("P");
        } else {
            Console.WriteLine("N");
        }
    }
}

File: c_like/premio.c
#include <stdio.h>

int main() {
    // Declaração das variáveis para pães (p), doces (d) e bolos (b)
    int p, d, b;
    int pontuacao_total;

    // Leitura dos dados de entrada
    scanf("%d", &p);
    scanf("%d", &d);
    scanf("%d", &b);

    // Processamento: Cálculo da pontuação conforme regras
    // Pão = 1 pt, Doce = 2 pts, Bolo = 3 pts
    pontuacao_total = (p * 1) + (d * 2) + (b * 3);

    // Lógica de Seleção para determinar a saída
    if (pontuacao_total >= 150) {
        printf("B\n"); // Ganha Bolo
    } else if (pontuacao_total >= 120) {
        printf("D\n"); // Ganha Doce
    } else if (pontuacao_total >= 100) {
        printf("P\n"); // Ganha Pão
    } else {
        printf("N\n"); // Não ganha prêmio
    }

    return 0;
}